File: claim-db/src/main/java/com/compalinttracker/claimdb/complaint/ComplaintDto.java
package com.compalinttracker.claimdb.complaint;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.apache.tomcat.jni.Local;
import org.springframework.format.annotation.DateTimeFormat;

import javax.validation.constraints.NotBlank;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ComplaintDto {

    //@NotBlank(message = "Serial number is mandatory")
    private String serialNumber;
    private String qmsNumber;
    private String customerRefNumber;
    private String claimedFault;
    private String productInfo;

    private String arrivedAt;
    private UUID responsible;
    private Boolean isPrio;
}

File: claim-frontend/src/app/complaint/complaintDto.ts
import { UserDto } from "../user/userDto";

export interface ComplaintDto {
   
    id: number;
    serialNumber: string;
    qmsNumber: string;
    customerRefNumber: string;
    claimedFault: string;
    arrivedAt: string;
    productInfo: string;
    responsible: string;
    responsibleId: string;

    analysisStartedAt: string;
    analysisEndedAt: string;

    faultVerification: boolean;

    barcodes: string;
    lifecycleInfo: string;
    visualAnalysis: string;
    electricalAnalysis: string;
    conclusion: string;

   // productGroup: string;
    analyzedBy: string;
    analyzedById: string;

    isPrio: boolean;
   
}


